### PR TITLE
Support text wrapping

### LINKED ISSUE #16: Support text wrapping
Hello, I get a box collapse when I want to print a string longer than the window width. Is it a normal behavior?

### Screenshot:

image

### Steps to reproduce:

```js
const boxen = require('boxen');
console.log(boxen('unicornddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddd', {padding: 1}));
```

### PR DESCRIPTION
**Pull request by lukaszmn**
I added support for wrapping text using `wrap-ansi`. It seems to be well tested and takes into consideration both ANSI codes and Unicode characters.

This should fix #16.

Closes #36

### CODE REVIEW

**Comment by sindresorhus** [comment]
Can you add some various example to https://github.com/sindresorhus/boxen/blob/master/example.js with text wrapping? So we can easily manually verify it works and also show the user that it works great.

File: test.js [hunk 1]
```diff
@@ -325,3 +325,170 @@ ${dimSide}foo${dimSide}
...
 	`);
 });
+
+test('no wrapping when content = columns - 2 and no padding and no margin', t => {
+	const width = process.stdout.columns;
+	const longContent = 'x'.repeat(width - 2);
+	const box = boxen(longContent);
+
+	// No endlines
+	t.is(box.indexOf('\n'), -1);
```
Comment (sindresorhus) [inline test.js]:
```suggestion
	t.false(box.includes('\n'));
```

Same with the other places this is done.

File: test.js [hunk 2]
```diff
@@ -325,3 +325,170 @@ ${dimSide}foo${dimSide}
...
+	const longContent = 'x'.repeat(width - 8);
+	// This gives only 3 spaces of margin on each side, but 5 were requested
+	const box = boxen(longContent, {margin: {left: 5, right: 5}});
+
+	const boxWidth = width - 3;
+	// Minus 2 NL
+	t.is(box.length - 2, 3 * boxWidth);
+
+	const lines = box.split('\n');
+	lines.forEach((line, index) => t.is(line.length, boxWidth, 'Length of line #' + index));
```
Comment (sindresorhus) [inline test.js]: Use a for-of loop. More readable.

**Comment by lukaszmn** [comment]
Sorry, I had to rebase to include changes from your branch. Besides I discovered a bug with alignment and fixed it and did all the changes you requested.

**Comment by sindresorhus** [comment]
Looks good now. Thank you 👍

**Comment by sindresorhus** [comment]
Hmm, a test is failing when merged to master: https://travis-ci.com/github/sindresorhus/boxen/jobs/328590367 Could you take a look at it?

**Pull request merged**